c/misra/src/rules/RULE-2-8/UnusedObjectDefinitionInMacroStrict.ql

```diff
@@ -0,0 +1,27 @@
+/**
+ * @id c/misra/unused-object-definition-in-macro-strict
+ * @name RULE-2-8: Project macros should not include '__attribute__((unused))' object definitions
+ * @description A strict query which reports all unused object definitions in macros with
+ *              '__attribute__((unused))'.
+ * @kind problem
+ * @precision very-high
+ * @problem.severity recommendation
+ * @tags external/misra/id/rule-2-8
+ *       maintainability
+ *       performance
+ *       external/misra/c/2012/amendment4
+ *       external/misra/c/strict
```

Comment:
I think it makes sense to use the tag "strict" here to cover these cases. Although we've added strict tags before (to AUTOSAR queries), we've never integrated them into our query suites, or provided user facing recommendations around them.

There are a few ways we could consider doing so, but I think my preference would be to:

 1. Exclude strict queries from the default query suites (for only MISRA C to start with - the AUTOSAR strict queries would need to be re-reviewed)
 2. Provide a strict query suite separately.
 3. Update the `user_manual.md` to discuss the option of "strict" mode.
 4. Update the `development_handbook.md` to discuss the criteria for a query to be considered for "strict" mode.

We should also, in this case, put into the "implementation_scope" why we consider this to be a "strict" query - i.e. why it is safe to default disable it.

The alternative to a "strict mode tag", that I think is worth mentioning here, is to provide a default "deviations" configuration that disables such queries that we would consider to be "strict".

Reply:
Offered a strict definition of what should be considered a strict query, hopefully it LGTY